crates/vm/src/stdlib/io.rs

```diff
@@ -2439,6 +2439,26 @@ mod _io {
             Ok(())
         }
 
+        #[pymethod]
+        fn detach(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult {
+            let mut textio = zelf.lock(vm)?;
+
+            // Fail fast if already detached
+            if vm.is_none(&textio.buffer) {
+                return Err(vm.new_value_error("underlying buffer has been detached"));
+            }
+
+            // Inline the equivalent of `flush` method without reentering the lock
+            textio.check_closed(vm)?;
+            textio.telling = textio.seekable;
+            textio.write_pending(vm)?;
+            vm.call_method(&textio.buffer, "flush", ())?;
```

Comment:
Could you create `fn flush_inner(textio: ..., vm: ...) -> PyResult` to share the actual flushing between methods?

@coderabbitai please advice about the nominal type of `textio`